Remotion/Data/DomainObjects/Mapping/Validation/Logical/StorageGroupTypesAreSameWithinInheritanceTreeRule.cs

```diff
@@ -21,20 +21,20 @@ namespace Remotion.Data.DomainObjects.Mapping.Validation.Logical
   /// <summary>
   /// Validates that the class has the same storage group type as its base class.
   /// </summary>
-  public class StorageGroupTypesAreSameWithinInheritanceTreeRule : IClassDefinitionValidationRule
+  public class StorageGroupTypesAreSameWithinInheritanceTreeRule : ITypeDefinitionValidationRule
   {
-    public MappingValidationResult Validate (ClassDefinition classDefinition)
+    public MappingValidationResult Validate (TypeDefinition typeDefinition)
     {
-      if (classDefinition.BaseClass != null)
+      if (typeDefinition is ClassDefinition classDefinition && classDefinition.BaseClass != null)
```

Comment:
this needs an interface-path, too. It tests that the baseclass and all implemented interfaces satisfy this rule